File: smash-ranks-android/app/src/main/java/com/garpr/android/adapters/RegionsSelectionAdapter.java
package com.garpr.android.adapters;

import android.content.Context;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;

import com.garpr.android.R;
import com.garpr.android.models.Region;
import com.garpr.android.models.RegionsBundle;

public class RegionsSelectionAdapter extends BaseAdapter<Region> {

    public RegionsSelectionAdapter(@NonNull final Context context) {
        super(context);
        setHasStableIds(true);
    }

    public RegionsSelectionAdapter(@NonNull final LayoutInflater layoutInflater) {
        super(layoutInflater);
        setHasStableIds(true);
    }

    @Override
    public long getItemId(final int position) {
        return getItem(position).hashCode();
    }

    @LayoutRes
    @Override
    public int getItemViewType(final int position) {
        return R.layout.item_region_selection;
    }

    public void set(@Nullable final RegionsBundle bundle) {
        if (bundle != null && bundle.hasRegions()) {
            set(bundle.getRegions());
        } else {
            clear();
        }
    }

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/adapters/RegionsSelectionAdapter.kt
package com.garpr.android.adapters

import com.garpr.android.R
import com.garpr.android.data.models.Endpoint
import com.garpr.android.data.models.Region

class RegionsSelectionAdapter : BaseMultiAdapter(LAYOUT_KEY_MAP) {

    companion object {
        private val LAYOUT_KEY_MAP = mapOf<Class<*>, Int>(
                Endpoint::class.java to R.layout.divider_endpoint,
                Region::class.java to R.layout.item_region_selection
        )
    }

    init {
        setHasStableIds(true)
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

}
